### PR TITLE
Adds support for rootDir in .clasp.json

### LINKED ISSUE #22: Possible to specify directory for push / pull ?
## Expected Behavior
I have my source files in an `src` directory, but do some transpiling which outputs the files to a `build` directory. Is it possible to specify a directory to push files from?

## Actual Behavior
Files are pushed and pulled from the root of the repo

## Specifications

  - Node version (`node -v`): any
  - Version (`npm list | grep clasp`): 1.0.7
  - OS (Mac/Linux/Windows): Mac

### PR DESCRIPTION
**Pull request by jondcallahan**
closes #22 

Solves for users who deploy from a non-root directory (ex. `build`) by adding support for a `rootDir` property in the project config file. This directory will be used in the `push` and `pull` commands. Defaults to the root directory when the property does not exist.

### CODE REVIEW

**Comment by jondcallahan** [comment]
I signed it!

**Comment by jondcallahan** [comment]
I signed it again!

**Comment by grant** [comment]
Great, I'll look at this later this week.

File: index.js [hunk 1]
```diff
@@ -482,10 +483,12 @@ program
               let nonIgnoredFilePaths = [];
               let files = filePaths.map((name, i) => {
                 let nameWithoutExt = name.slice(0, -path.extname(name).length);
+                // Formats rootDir/appsscript.json to appsscript.json. Preserves subdirectory names in rootDir
```
Comment (grant) [inline index.js]: Please wrap the comment.

File: index.js [hunk 2]
```diff
@@ -482,10 +483,12 @@ program
...
+                let formattedName = rootDir ? formattedName.slice(rootDir.length + 1, formattedName.length) : nameWithoutExt;
```
Comment (grant) [inline index.js]: Please wrap this long line.

File: index.js [hunk 3]
```diff
@@ -411,13 +412,13 @@ function fetchProject(scriptId) {
...
-          let truePath = `./${filePath}`;
+          let truePath = `${rootDir || '.'}/${filePath}`;
           mkdirp(path.dirname(truePath), (err) => {
             if (err) return logError(err, ERROR.FS_DIR_WRITE);
-            fs.writeFile(`./${filePath}`, file.source, (err) => {
+            fs.writeFile(truePath, file.source, (err) => {
               if (err) return logError(err, ERROR.FS_FILE_WRITE);
             });
-            console.log(`└─ ${filePath}`);
+            console.log(`└─ ${rootDir ? truePath : filePath}`); // Log only filename if pulling to root (Code.gs vs ./Code.gs)
```
Comment (grant) [inline index.js]: Please wrap.

File: index.js [hunk 4]
```diff
@@ -444,10 +445,10 @@ program
   .command('pull')
   .description('Fetch a remote project')
   .action(() => {
-    getProjectSettings().then((scriptId) => {
+    getProjectSettings().then(({scriptId, rootDir}) => {
```
Comment (grant) [inline index.js]: Please add spaces when destructuring:

`getProjectSettings().then(({ scriptId, rootDir }) => {`

File: index.js (hunk 2 above)
Comment (grant) [reply index.js]: How do you declare `formattedName` and use it in the same statement? Do you mean `nameWithoutExt`? Maybe I'm missing something.

**Comment by jondcallahan** [comment]
Thanks for the feedback @grant - addressed your comments in the latest commit.

**Pull request merged**